evalme/eval_item.py

```diff
@@ -31,3 +31,139 @@ def empty(self):
 
     def __len__(self):
         return len(self._raw_data)
+
+    @staticmethod
+    def has_spans(results_list):
+        for r in results_list:
+            if not 'start' in r or not 'end' in r:
+                return False
+        return True
+
+    @staticmethod
+    def spans_iou(x, y):
+        """
+        Intersection over union for spans with start/end
+        :param x:
+        :param y:
+        :return:
+        """
+        assert EvalItem.has_spans([x, y])
+        s1, e1 = float(x['start']), float(x['end'])
+        s2, e2 = float(y['start']), float(y['end'])
+        if s2 > e1 or s1 > e2:
+            return 0
+        intersection = min(e1, e2) - max(s1, s2)
+        union = max(e1, e2) - min(s1, s2)
+        if union == 0:
+            return 0
+        iou = intersection / union
+        return iou
+
+    @staticmethod
+    def bbox_iou(boxA, boxB):
+        """
+        Intersection over union for bbox with x/y/width/height
+        :param boxA:
+        :param boxB:
+        :return:
+        """
+        # check data
+        assert boxA['x']
```

Comment:
that gives an error when boxA['x'] == 0 and exception with non-existed key

Reply:
Fixed